api/apipb/api.proto

```diff
@@ -0,0 +1,164 @@
+	// 1. start index and action count
+	// 2. action hash
+	// 3. address with start index and action count
+	// 4. get unconfirmed actions by address with start index and action count
+	// 5. block hash with start index and action count
+	rpc GetActions(GetActionsRequest) returns (GetActionsResponse) {}
+
+    // get block metadata(s) by:
+	// 1. start index and block count
+	// 2. block hash
+	rpc GetBlockMetas(GetBlockMetasRequest) returns (GetBlockMetasResponse) {}
+
+    // get chain metadata
+	rpc GetChainMeta(GetChainMetaRequest) returns (GetChainMetaResponse) {}
+
+    // sendAction
+	rpc SendAction(SendActionRequest) returns (SendActionResponse) {}
+
+    // get receipt by action Hash
+	rpc GetReceiptByAction(GetReceiptByActionRequest) returns (GetReceiptByActionResponse) {}
+
+    // TODO: read contract
+	rpc ReadContract(ReadContractRequest) returns (ReadContractResponse) {}
+
+    // suggest gas price
+	rpc SuggestGasPrice(SuggestGasPriceRequest) returns (SuggestGasPriceResponse) {}
+
+    // estimate gas for action
+	rpc EstimateGasForAction(EstimateGasForActionRequest) returns (EstimateGasForActionResponse) {}
+}
+
+message GetAccountRequest {
+	string address = 1;
+}
+
+message GetAccountResponse {
+	iproto.AccountMeta accountMeta = 1;
+}
+
+message GetActionsRequest {
+	oneof request {
+		GetActionsByIndexRequest byIndex = 1;
+		GetActionByHashRequest byHash = 2;
+		GetActionByHashRequest byAddr = 3;
+		GetUnconfirmedActionsByAddressRequest unconfirmedByAddr = 4;
+		GetActionsByBlockRequest byBlk = 5;
+	}
+}
+
+message GetActionsByIndexRequest {
+	uint64 start = 1;
+	uint64 count = 2;
+}
+
+message GetActionByHashRequest {
+	string actionHash = 1;
+	bool checkPending = 2;
+}
+
+message GetActionByAddressRequest {
+	string address = 1;
+	uint64 start = 2;
+	uint64 count = 3;
+}
+
+message GetUnconfirmedActionsByAddressRequest {
+	string address = 1;
+	uint64 start = 2;
+	uint64 count = 3;
+}
+
+message GetActionsByBlockRequest {
+	string blkHash = 1;
+	uint64 start = 2;
+	uint64 count = 3;
+}
+
+message GetActionsResponse {
+	repeated iproto.ActionPb actions = 1;
+}
+
+message GetBlockMetasRequest {
+	oneof request {
+		GetBlockMetasByIndexRequest byIndex = 1;
+		GetBlockMetaByHashRequest byHash = 2;
+	}
+}
+
+message GetBlockMetasByIndexRequest {
+	uint64 start = 1;
+	uint64 count = 2;
+}
+
+message GetBlockMetaByHashRequest {
+	string blkHash = 1;
+}
+
+message GetBlockMetasResponse {
+	repeated iproto.BlockMeta blkMetas = 1;
+}
+
+message GetChainMetaRequest {}
+
+message GetChainMetaResponse {
+	iproto.ChainMeta chainMeta = 1;
+}
+
+message SendActionRequest {
+	iproto.ActionPb action = 1;
+}
+
+message SendActionResponse {
+	string actionHash = 1;
+}
+
+message GetReceiptByActionRequest {
+	string actionHash = 1;
+}
+
+message GetReceiptByActionResponse {
+	iproto.ReceiptPb receipt = 1;
+}
+
+message ReadContractRequest {
+	iproto.ActionPb action = 1;
+}
+
+message ReadContractResponse {
+	string hash = 1;
```

Comment:
data?

Reply:
This response should be receipt's return value. We can come up with a different name.